crates/plugin-engine/src/internal/lualib/init.rs

```diff
@@ -0,0 +1,23 @@
+use mlua::{Lua, Value};
+
+use super::{
+    encoding::LuaEncodingLib, hashing::LuaHashingLib, logger::LuaLogger, strings::LuaStringsLib,
+    tables::LuaTablesLib,
+};
+
+// TODO: return and propagate a Result type
+pub fn new_lua_environment() -> &'static Lua {
+    let lua = Box::leak(Box::new(Lua::new()));
+
+    let lldap_lib = lua.create_table().unwrap();
+    let _ = lldap_lib.set("encoding", LuaEncodingLib {});
+    let _ = lldap_lib.set("hashing", LuaHashingLib {});
+    let _ = lldap_lib.set("log", LuaLogger {});
+    let _ = lldap_lib.set("strings", LuaStringsLib {});
+    let _ = lldap_lib.set("tables", LuaTablesLib {});
```

Comment:
This is where we'll add functionality for calling Http endpoints, I guess?

Reply:
Yes, that was what I was thinking.